CodeEdit/SideBar/SideBarItem.swift

```diff
@@ -43,17 +43,15 @@ struct SideBarItem: View {
 
 	func sidebarFolderItem(_ item: WorkspaceClient.FileItem) -> some View {
 		DisclosureGroup(isExpanded: $isExpanded) {
-			if isExpanded { // Only load when expanded -> Improves performance massively
 				ForEach(item.children!.sortItems(foldersOnTop: workspace.sortFoldersOnTop)) { child in
```

Comment:
This is actually required since in a very large project folder this will load all children all the time which leads to very long file opening times.

Reply:
https://user-images.githubusercontent.com/9460130/159027490-f99ef0e6-6af5-46d1-ae57-38a4aab6598a.mov

Reply:
It breaks the animation.

Maybe it's an idea to only place the labels, and then update them?

Reply:
If you can figure something out that doesn't break performance that'd be awesome. So far this `if` clause was the best option I found

Reply:
So instead of 
```swift
SideBarItem(item: child,
								workspace: workspace,
                                windowController: windowController)
```,

```swift
SideBarItem(item: child,
								workspace: workspace,
                                windowController: windowController,
                                loadSubitems: isExpanded)
```

Or (maybe the worst solution)
```
Text(child.name) // Don't know is child.name exists. its pseudo code
```

Reply:
Try opening a node project with dozens of dependencies to test performance.

Reply:
@lukepistrol can you please try the last commit?

```swift
		DisclosureGroup(isExpanded: $isExpanded) {
			if !isExpanded {
                // Only load the labels
                ForEach(item.children!.sortItems(foldersOnTop: workspace.sortFoldersOnTop)) { child in
                    Label(child.url.lastPathComponent, systemImage: child.systemImage)
                        .accentColor(child.iconColor)
                        .font(.callout)
                }
            } else {
                // Only load when expanded -> Improves performance massively
				ForEach(item.children!.sortItems(foldersOnTop: workspace.sortFoldersOnTop)) { child in
					SideBarItem(item: child,
								workspace: workspace,
                                windowController: windowController)
				}
			}
```

Reply:
Will try in a few minutes

Reply:
Ah, I see, the issue is, when you open a file, it's rendering the whole file tree again.

Reply:
That's what I suspected as well

Reply:
ok I just tested the latest commit and it seems to run very nice. Couldn't see performance losses to before.